src/app/api/cronjob/route.ts

```diff
@@ -0,0 +1,177 @@
+        ),
+        gte(semesters.endDate, format(currentDateUTC, "yyyy-MM-dd")),
+      ),
+    });
+
+    if (!activeSemester) {
+      return NextResponse.json("No active semester found.", { status: 400 });
+    }
+
+    //const semesterStartDate = new Date(activeSemester.startDate);
+    const semesterEndDate = new Date(activeSemester.endDate);
+    const semesterBreakStart = new Date(activeSemester.breakStart);
+    const semesterBreakEnd = new Date(activeSemester.breakEnd);
+    const bookingOpenDay = activeSemester.bookingOpenDay;
+    const diff = findDiff(now, bookingOpenDay);
+    const startDate = addDays(currentDateUTC, diff);
+
+    // Check if the next batch of sessions is within the semester break or after the end of semester
+    if (startDate > semesterEndDate) {
+      return NextResponse.json(
+        "No sessions were generated as they are all after the semester end date.",
+        {
+          status: 400,
+        },
+      );
+    }
+
+    if (
+      addDays(startDate, 6) <= semesterBreakEnd &&
+      startDate >= semesterBreakStart
+    ) {
+      return NextResponse.json(
+        "No sessions were generated as they are all within the semester break.",
+        {
+          status: 400,
+        },
+      );
+    }
+
+    await db.transaction(async (tx) => {
+      // For each day in the session week, create a game session
+      for (let i: number = 0; i < 7; i++) {
+        const day = addDays(startDate, i);
+
+        if (day > semesterEndDate) {
+          break;
+        }
+
+        if (day >= semesterBreakStart && day <= semesterBreakEnd) {
+          continue;
+        }
+
+        //check for gameSessionScheduleExceptions and continue if deleted, and if edited then insert that instead and continue
+
+        // Find game session schedule for the day
+        const schedule = await tx.query.gameSessionSchedules.findFirst({
+          where: and(
+            eq(gameSessionSchedules.weekday, getWeekday(day)),
+            eq(gameSessionSchedules.semesterId, activeSemester.id),
+          ),
+        });
+
+        if (!schedule) {
+          continue;
+        }
+
+        // Calculate booking close time
+        const [hours, mins, secs] = schedule.startTime.split(":").map(Number);
+        const bookingClose = new Date(
+          day.getTime() + 3600 * 1000 * hours + 60 * 1000 * mins + 1000 * secs,
+        );
+
+        const insertSchedule = insertGameSessionSchema.parse({
+          bookingOpen: startDate,
+          bookingClose: bookingClose,
+          gameSessionScheduleId: schedule.id,
+          date: format(day, "yyyy-MM-dd"),
+          startTime: schedule.startTime,
+          endTime: schedule.endTime,
+          locationName: schedule.locationName,
+          locationAddress: schedule.locationAddress,
+          capacity: schedule.capacity,
+          casualCapacity: schedule.casualCapacity,
+        });
+
+        await tx.insert(gameSessions).values(insertSchedule);
```

Comment:
Only insert if there isn't a existing game session for that date.

Use .onConflictDoNothing()